### DISCUSSION TITLE
Warning when installing[tensorflow]

### DISCUSSION THREAD

**Discussion by ShuiqingHuang** [asreview/asreview#1672] [Troubleshooting (report errors in the "Issues" tracker)]
I have tried installing python 3.9-3.11 to install ASReview and tensorflow, but when I launched ASReview, I would get the warning like this:

X:\>asreview lab
2024-02-01 14:29:16.214072: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
WARNING:tensorflow:From C:\Users\P314638\AppData\Local\Programs\Python\Python311\Lib\site-packages\keras\src\losses.py:2976: The name tf.losses.sparse_softmax_cross_entropy is deprecated. Please use tf.compat.v1.losses.sparse_softmax_cross_entropy instead.

The ASReview can be opened and used normally with this warning, but in order to solve this warning, what could I do? Or are there potential risks regarding this warning?

**Comment by jteijema** [comment]
Hello @ShuiqingHuang ,

The warning you've noticed is actually from Keras, a neural networks API that ASReview relies on, which in turn is using a deprecated TensorFlow function. This situation is typical when libraries are updated at different times. Despite the warning, it does not impact ASReview's functionality and can be safely ignored. I assume a following update of Keras will solve this warning!

As for the oneDNN option; tensorflow is optimized for using your hardware as fast as possible. Every system is different, and apparently yours supports oneDNN options that work best with floating-point round-offs. This should give no problems, aside from the possibility that the nn-2-layer model will output a VERY slightly different rank order than on other systems. Not to worry though, asreview projects store the output at every step along the way, meaning retracing your steps is still possible in a scientific environment.

Best,
^JT
